src/phylum/ci/ci_gitlab.py

```diff
@@ -124,6 +125,11 @@ def common_ancestor_commit(self) -> Optional[str]:
             default_branch_name = git_default_branch_name(remote)
         default_branch = f"refs/remotes/{remote}/{default_branch_name}"
 
+        project_dir = Path(os.getenv("CI_PROJECT_DIR", ".")).resolve()
+        if not Path(project_dir, ".git", default_branch).resolve().exists():
+            LOG.warning("The default remote branch is not available. Attempting to fetch it...")
+            git_fetch(repo=remote, ref=default_branch_name, git_c_path=project_dir)
```

Comment:
Only complaint I could see for this is that it relies very much on git internals. In theory using git's CLI would be a more robust interface.

Reply:
The `git_fetch` function does use the git CLI.

Reply:
Just to be clear, I was talking about the way the path is checked to ensure the rev is available.

Reply:
Oooh...yeah...that makes more sense. Sorry for the misunderstanding. Yes, this can also be written with the git CLI with something like:

`git show-ref --quiet --verify f"{default_branch}"`

A note has been taken and this will be updated going forward.